### ISSUE TITLE
Revisit record_assessor parsing or documentation explanation about it.

### ISSUE THREAD

**Issue by divbell** [fluent/fluent-bit#6751]
**Is your feature request related to a problem? Please describe.**

I'm frustrated by the inability to include a slash character after a variable in the log_group_template.

The documentation indicates the following:
```Notice in the example above, that the template values are separated by dot characters. This is important; the Fluent Bit record_accessor library has a limitation in the characters that can separate template variables- only dots and commas (. and ,) can come after a template variable. This is because the templating library must parse the template and determine the end of a variable.```

**Describe the solution you'd like**

This explanation suggests it is not possible to parse the template and determine the end of a variable.  I'm fairly certain though that one could exclude a slash as a part of a variable or this could otherwise be parsed.

If nothing else, a better explanation about why it is a limitation would be helpful.  The existing explanation "_because the templating library must parse the template and determine the end of a variable_" is not indicative of why this would not be possible, other than it being a simplistic parsing implementation.

**Describe alternatives you've considered**

Basically I'm stuck on the legacy "cloudwatch" output because of this but I want to use the "cloudwatch_logs" version.

**Additional context**

This affects me by being unable to hierarchically nest cloudwatch log groups by namespace and pod name.

This is the context of how I want to store the logs into a templated log group name:

`/aws/containerinsights/${CLUSTER_NAME}/$(kubernetes['namespace_name'])/$(kubernetes['labels']['podname'])/app
lication`

**Comment by divbell** [comment]
It occurs to me that I could build up the value I desire in one variable.
